src/transformers/modeling_utils.py

```diff
@@ -428,6 +429,10 @@ def load_sharded_checkpoint(model, folder, strict=True, prefer_safe=True):
             - `missing_keys` is a list of str containing the missing keys
             - `unexpected_keys` is a list of str containing the unexpected keys
     """
+    logger.warning(
+        "Note that `load_sharded_checkpoint` is deprecated and will be removed in v4.50. We recommend you using "
```

Comment:
most likely in `transformers 5.0` => transformers is more conservative than `huggingface_hub` in general